pkg/component/runtime/manager.go

```diff
@@ -694,6 +765,50 @@ func (m *Manager) getListenAddr() string {
 	return m.listenAddr
 }
 
+func (m *Manager) performDiagAction(ctx context.Context, unit component.Unit) ([]*proto.ActionDiagnosticUnitResult, error) {
+	ctx, cancel := context.WithTimeout(ctx, diagnosticTimeout)
+	defer cancel()
+
+	id, err := uuid.NewV4()
+	if err != nil {
+		return nil, err
+	}
+
+	runtime := m.getRuntimeFromUnit(unit)
+	if runtime == nil {
+		return nil, ErrNoUnit
+	}
+
+	req := &proto.ActionRequest{
+		Id:       id.String(),
+		UnitId:   unit.ID,
+		UnitType: proto.UnitType(unit.Type),
+		Type:     proto.ActionRequest_DIAGNOSTICS,
+	}
+	res, err := runtime.performAction(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	if res.Status == proto.ActionResponse_FAILED {
+		var respBody map[string]interface{}
+		if res.Result != nil {
+			err = json.Unmarshal(res.Result, &respBody)
+			if err != nil {
+				return nil, err
+			}
+			errMsgT, ok := respBody["error"]
+			if ok {
+				errMsg, ok := errMsgT.(string)
+				if ok {
+					return nil, errors.New(errMsg)
+				}
+			}
+		}
+		return nil, errors.New("generic action failure")
```

Comment:
[Suggestion]
Please be more specific. If I understand it correctly the unit failed to perform the diagnostics, but still returned a response. However no error could be extracted from it. Thus I'd ask to inform that in the error. Perhaps something like 
```suggestion
		return nil, errors.New("unit failed to perform diagnostics, no error could be extracted from response")
```

Another thing is, in this all failed case, wouldn't be good to log the response so it can be investigated later? It even could be that the error is indeed in the response, but it's malformed, so we could not extract it.